### PR TITLE
EZP-29820: /user/update/{id}/{version}/{language} access limitation

### PR DESCRIPTION
**Pull request by ViniTou**
| Question      | Answer
| ------------- | ---
| Tickets       | https://jira.ez.no/browse/EZP-29820
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    |no
| Tests pass?   | yes
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

### CODE REVIEW

File: bundle/Controller/UserController.php [hunk 1]
```diff
@@ -149,6 +156,9 @@ public function editAction(
         Request $request
     ) {
         $user = $this->userService->loadUser($contentId);
+        if (!$this->permissionResolver->canUser('content', 'read', $user)) {
+            throw new CoreUnauthorizedException('content', 'read', array('userId' => $contentId));
```
Comment (adamwojs) [inline bundle/Controller/UserController.php]:
```suggestion
            throw new CoreUnauthorizedException('content', 'read', ['userId' => $contentId]);
```

Comment (mikadamczyk) [inline bundle/Controller/UserController.php]: I think we should use `eZ\Publish\API\Repository\Exceptions\UnauthorizedException`

**Comment by ViniTou** [comment]
@andrerom 
Do you mean something similar inside `\eZ\Publish\Core\Repository\UserService::updateUser`
```
        $canEditContent = $this->permissionResolver->canUser('content', 'edit', $loadedUser);

        if (!$canEditContent && $this->isUserProfileUpdateRequested($userUpdateStruct)) {
            throw new UnauthorizedException('content', 'edit');
        }
```

**Comment by lserwatka** [comment]
Yes, it should check content/edit

File: bundle/Controller/UserController.php [hunk 2]
```diff
@@ -149,6 +156,9 @@ public function editAction(
         Request $request
     ) {
         $user = $this->userService->loadUser($contentId);
+        if (!$this->permissionResolver->canUser('content', 'read', $user)) {
+            throw new CoreUnauthorizedException('content', 'read', ['userId' => $contentId]);
```
Comment (glye) [inline bundle/Controller/UserController.php]:
```suggestion
        if (!$this->permissionResolver->canUser('content', 'edit', $user)) {
            throw new CoreUnauthorizedException('content', 'edit', ['userId' => $contentId]);
```

Comment (glye) [inline bundle/Controller/UserController.php]:
```suggestion
            throw new CoreUnauthorizedException('content', 'edit', ['userId' => $contentId]);
```
Same on line above (seems we can't do multiline suggesions).

**Comment by glye** [comment]
Thanks @ViniTou. I have asked QA to do their thing.

**Comment by glye** [comment]
Thanks @micszo!
Everyone, any objections before we merge and tag this?

**Pull request merged**